### DISCUSSION TITLE
grod of twinned grains

### DISCUSSION THREAD

**Discussion by tongzhangqu** [mtex-toolbox/mtex#1737] [Ask Anything]
Hi, 

I am trying to have a grod map for grains with twinned sections. 

Part of my code is here:
```
%% Merge
[mergedGrains,parentId] = merge(grains, etwinBoundary); 
ebsd_merged = ebsd; % make a copy of ebsd data to a new variable
ebsd_merged.grainId = parentId(grains.id2ind(ebsd.grainId));% update the grainIds to the parentIds in ebsd_merged

%% Grain Reference Orientation Deviation (GROD)
F = halfQuadraticFilter; % establish a filter
F.alpha = 0.5;
%ebsd_merged = smooth(ebsd_merged,F,'fill',mergedGrains); %denoise the orientation map
grod = calcGROD(ebsd_merged, mergedGrains);

plot(ebsd_merged, grod.angle./degree,'micronbar','off','threshold',20*degree)
mtexColorbar('title','misorientation angle to meanorientation in degree')
mtexColorMap LaboTeX
hold on
plot(mergedGrains.boundary, 'linewidth', 2);
plot(etwinBoundary,'linecolor','blue', 'linewidth', 1.5, 'figSize','normal');
hold off
```

I have also tried this to calculate grod. 
```
meanOri = mergedGrains.meanOrientation
meanOri = meanOri(ebsd_merged.grainId)
grod = inv(meanOri) .* ebsd_merged.orientations

```

The results are the same. There is no grod in all the merged grains containing twin boundaries in blue. Is there a way to calculate grod in grains containing the twin boundaries?
![image](https://user-images.githubusercontent.com/107702654/218718045-4abd0edf-69ab-4f13-9832-da907705fee8.png)

Thank you,
Tongzhang

**Comment by kilir** [comment]
Hi,
since merge() does not produce grains.meanOrientation (since it is most likely non-sensical for a twinned grain), I guess those are set to nan. You can enforce the computation of grains.meanOrientation using the option 'calcMeanOrientation', but one could wonder about the meaning of the result. Maybe you could something more reasonable than the mean orientation as a reference orientation in calcGROD.

Cheers,
Rüdiger

**Reply by tongzhangqu** [reply]
Hi Rüdiger
Thank you so much for the replying. You are right. Mean orientation for a twinned grain is meaningless. Perhaps KAM is better to reveal the local variation of orientations.
Thank you.
Tongzhang
